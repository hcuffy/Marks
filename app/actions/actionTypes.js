export const HANDLE_SCHOOL_DATA = 'HANDLE_SCHOOL_DATA';
export const HANDLE_SCHOOL_DATA_DISPLAY = 'HANDLE_SCHOOL_DATA_DISPLAY';
