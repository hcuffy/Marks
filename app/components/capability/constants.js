export const OPEN_CLOSE_CLASS_LIST = 'OPEN_CLOSE_CLASS_LIST'
export const OPEN_CLOSE_STUDENT_LIST = 'OPEN_CLOSE_STUDENT_LIST'
export const GET_ALL_ANSWERS = 'GET_ALL_ANSWERS'
export const OPEN_CLOSE_QUESTION_LIST = 'OPEN_CLOSE_QUESTION_LIST'
export const UPDATE_QUESTION_SET = 'UPDATE_QUESTION_SET'
