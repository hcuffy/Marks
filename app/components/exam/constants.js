export const GET_SELECTED_CLASS = 'GET_SELECTED_CLASS'
export const ADD_NEW_EXAM = 'ADD_NEW_EXAM'
export const UPDATE_DROPDOWN_CLASS_LIST = 'UPDATE_DROPDOWN_CLASS_LIST'
export const DISPLAY_SUBJECT_LIST = 'DISPLAY_SUBJECT_LIST'
export const GET_SINGLE_EXAM = 'GET_SINGLE_EXAM'
export const UPDATE_EXAMS_LIST = 'UPDATE_EXAMS_LIST'
export const EXAM_FORM_VALIDATION = 'EXAM_FORM_VALIDATION'
