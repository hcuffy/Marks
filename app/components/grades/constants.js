export const DISPLAY_EXAM_TABLE = 'DISPLAY_EXAM_TABLE'
export const OPEN_CLASS_LIST = 'OPEN_CLASS_LIST'
export const UPDATE_EXAM_TABLE = 'UPDATE_EXAM_TABLE'
