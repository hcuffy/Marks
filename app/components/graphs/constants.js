export const OPEN_GRAPH_CLASS_LIST = 'OPEN_GRAPH_CLASS_LIST'
export const GET_ALL_GRADES = 'GET_ALL_GRADES'
export const DISPLAY_SUBJECT_GRADES = 'DISPLAY_SUBJECT_GRADES'
export const DISPLAY_EXAM_GRADES = 'DISPLAY_EXAM_GRADES'
export const GET_ALL_EXAMS = 'GET_ALL_EXAMS'
