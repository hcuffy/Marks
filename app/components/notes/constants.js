export const GET_ALL_NOTES = 'GET_ALL_NOTES'
export const OPEN_STUDENT_NOTES_DROPDOWN = 'OPEN_STUDENT_NOTES_DROPDOWN'
export const OPEN_NOTES_DROPDOWN = 'OPEN_NOTES_DROPDOWN'
export const UPDATE_TEXTAREA = 'UPDATE_TEXTAREA'
export const UPDATE_TITLE = 'UPDATE_TITLE'
export const CLEAR_NOTE_FIELDS = 'CLEAR_NOTE_FIELDS'
export const UPDATE_NOTE = 'UPDATE_NOTE'
export const NOTES_FORM_VALIDATION = 'NOTES_FORM_VALIDATION'
