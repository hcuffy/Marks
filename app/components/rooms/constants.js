export const CHANGE_CLASSROOM_TAB = 'CHANGE_CLASSROOM_TAB'
export const ADD_CLASSROOM_DATA = 'ADD_CLASSROOM_DATA'
export const GET_CLASSROOM_DATA = 'GET_CLASSROOM_DATA'
export const UPDATE_CLASSROOM = 'UPDATE_CLASSROOM'
export const OPEN_CLOSE_ROOM_MODAL = 'OPEN_CLOSE_ROOM_MODAL'
export const UPDATE_SUBJECT = 'UPDATE_SUBJECT'
export const CLASSROOM_FORM_VALIDATION = 'CLASSROOM_FORM_VALIDATION'
