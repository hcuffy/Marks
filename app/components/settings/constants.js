export const actions = {
	UPDATE_GRADING_DATA: 'UPDATE_GRADING_DATA',
	GET_SYSTEM_TYPE: 'GET_SYSTEM_TYPE',
	HANDLE_SCHOOL_DATA: 'HANDLE_SCHOOL_DATA',
	DISPLAY_SCHOOL_DATA: 'DISPLAY_SCHOOL_DATA'
}
