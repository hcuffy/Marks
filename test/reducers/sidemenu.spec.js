import applyMenuStyling from '../../app/reducers/sidemenuReducer'
import { HANDLE_MENU_CHANGE } from '../../app/constants/actionTypes'

const initialLoadState = {
	home: '#1dbb90',
	classroom: '',
	students: '',
	exams: '',
	graphs: '',
	settings: ''
}

describe('test sidemenu reducer', () => {
	it('should use default state', () => {
		expect(applyMenuStyling(initialLoadState, {})).toMatchSnapshot()
	})

	it('should handle menu change', () => {
		const styleUpdate = {
			home: '',
			classroom: '',
			students: '#1dbb90',
			exams: '',
			graphs: '',
			settings: ''
		}
		const payload = { styleUpdate }
		expect(applyMenuStyling({}, { type: HANDLE_MENU_CHANGE, payload })).toMatchSnapshot()
	})
})
