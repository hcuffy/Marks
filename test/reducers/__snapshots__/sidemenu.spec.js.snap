// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test sidemenu reducer should handle menu change 1`] = `
Object {
  "classroom": "",
  "exams": "",
  "graphs": "",
  "home": "",
  "settings": "",
  "students": "#1dbb90",
}
`;

exports[`test sidemenu reducer should use default state 1`] = `
Object {
  "classroom": "",
  "exams": "",
  "graphs": "",
  "home": "#1dbb90",
  "settings": "",
  "students": "",
}
`;
